R/reportEDGETransport.R

```diff
@@ -295,12 +295,11 @@ reportEDGETransport <- function(output_folder=".",
     prodFosH2 <- unique(prodFosH2)
 
     ## SE to be devoted to transport
-    prodFe <- readgdx(gdx, "vm_prodFe")[, value := value*TWa_2_EJ] ## in EJ
-    setnames(prodFe, c("year", "region", "se", "fe", "te", "value"))
-    prodFe[, sec := ifelse(fe %in% c("fepet", "fedie", "feelt", "fegat", "feh2t"), "trsp", NA)]
-    prodFe[, sec := ifelse(fe %in% c("fesos", "fehos", "feels", "feh2s", "fegas", "fehes"), "st", sec)]
-    prodFe = prodFe[,.(value = sum(value)), by = c("year", "region", "se", "sec")]
-    shareSe <- prodFe[, sharesec := value/sum(value), by = c("year", "region", "se")]
+    demSe[, sec := ifelse(fe %in% c("fepet", "fedie", "feelt", "fegat", "feh2t"), "trsp", NA)]
+    demSe[, sec := ifelse(fe %in% c("fesos", "fehos", "feels", "feh2s", "fegas", "fehes"), "st", sec)]
+    demSe = demSe[fe != "seh2",] ## remove electricity used to produce hydrogen
+    demSe = demSe[,.(value = sum(value)), by = c("year", "region", "se", "sec")]
+    shareSe <- demSe[, sharesec := value/sum(value), by = c("year", "region", "se")]
```

Comment:
To me it looks like synfuel share of hydrogen is not contained in sharesec for `H2` and `trsp`. There are only production pathways to FEs, not to `seliqfos`.

Reply:
This would require to add the share of `seliqfos` to transport (this you have) times the share of hydrogen production for synfuels if I'm not mistaken.